runtime/src/main/java/io/novafoundation/nova/runtime/storage/source/query/BaseStorageQueryContext.kt

```diff
@@ -62,7 +62,7 @@ abstract class BaseStorageQueryContext(
         vararg prefixArgs: Any?,
         keyExtractor: (StorageKeyComponents) -> K,
         binding: DynamicInstanceBinderWithKey<K, V>,
-        onDecodeException: (Exception) -> Unit
+        recover: (exception: Exception, rawValue: String?) -> Unit
```

Comment:
I suggest to make `recover` return an element of type `V` so we can potentially provide fallbacks for failed items in the client code